core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -171,17 +175,40 @@ class GroupCoordinator(val brokerId: Int,
         responseCallback(joinError(JoinGroupRequest.UNKNOWN_MEMBER_ID, Errors.UNKNOWN_MEMBER_ID))
       } else if (!group.supportsProtocols(protocolType, MemberMetadata.plainProtocolSet(protocols))) {
         responseCallback(joinError(JoinGroupRequest.UNKNOWN_MEMBER_ID, Errors.INCONSISTENT_GROUP_PROTOCOL))
+      } else if (group.hasStaticMember(groupInstanceId)) {
+        val memberId = group.getStaticMemberId(groupInstanceId)
```

Comment:
Hmm.. I thought the proposal called for generation of a new memberId when a static consumer is restarted. The purpose is to fence the old static member. How do we avoid two static members from being active at the same time? Perhaps I'm missing something?

Reply:
Good catch! We have slightly diverged from the original proposal, so that we no longer kick off rebalance when static member rejoins with unknown member id. Thus the generation could not be used to fence against duplicate static members. Will address this problem by replacing with a new member id.